### PR TITLE
add collapse

### PR DESCRIPTION
**Pull request by baiyaaaaa**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's [Contributing Guide](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.md).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.
#627

### CODE REVIEW

File: examples/docs/zh-CN/collapse.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+### Collapse Attributes
+| 参数      | 说明          | 类型      | 可选值                           | 默认值  |
+|---------- |-------------- |---------- |--------------------------------  |-------- |
+| accordion | 是否手风琴模式 | boolean | — | false |
+| value | 当前激活的面板 | string/array | — | — |
+
+### Collapse Events
+| 事件名称 | 说明 | 回调参数 |
+|---------|---------|---------|
+| change | 当前激活面板改变时触发 | (activeNames: array) |
```
Comment (Leopoldthecoder) [inline examples/docs/zh-CN/collapse.md]: 是不是要说一下，如果是手风琴模式，那么回调参数就是一个 string/number，而不是数组？

File: packages/collapse/src/collapse-item.vue [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        contentHeight: 0
+      };
+    },
+
+    props: {
+      title: String,
+      name: {
+        type: [String, Number],
+        default() {
+          return uid();
```
Comment (QingWei-Li) [inline packages/collapse/src/collapse-item.vue]: 直接取 this._uid 呗
image

File: packages/collapse/src/collapse-item.vue [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    methods: {
+      open() {
+        const contentElm = this.$refs.content;
+        const contentStyle = this.contentStyle;
+
+        contentStyle.display = 'block';
+        this.$nextTick(_ => {
+          contentStyle.height = this.contentHeight + 'px';
+          once(contentElm, 'transitionend', () => {
```
Comment (QingWei-Li) [inline packages/collapse/src/collapse-item.vue]: transitionEnd 不是还有其他前缀的么

File: examples/docs/zh-CN/collapse.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+### Collapse Attributes
+| 参数      | 说明          | 类型      | 可选值                           | 默认值  |
+|---------- |-------------- |---------- |--------------------------------  |-------- |
+| accordion | 是否手风琴模式 | boolean | — | false |
+| value | 当前激活的面板(如果是手风琴模式，绑定值类型需要为`string`，否则为`array`) | string/array | — | — |
+
+### Collapse Events
+| 事件名称 | 说明 | 回调参数 |
+|---------|---------|---------|
+| change | 当前激活面板改变时触发(如果是手风琴模式，参数 `activeNames` 类型为`string`，否则为`array`) | (activeNames: array\|array) |
```
Comment (Leopoldthecoder) [inline examples/docs/zh-CN/collapse.md]: `activeNames: array\|array`🌚

**Pull request merged**